### ISSUE TITLE
Can't Search By objectGUID using MOCK_SYNC

### ISSUE THREAD

**Issue by mprahl** [cannatag/ldap3#575]
Hello,
I have unit tests that tests a function's ability to search Active Directory via objectGUID. The JSON file describing the server is located at:
https://github.com/StackFocus/PostMaster/blob/master/tests/ad/ad_ldap_directory.json

When searching via objectGUID using Python 2.7, I get the following:
```python
>>> import ldap3
>>> mock_server = ldap3.Server('ldaps://server.domain.local:636', get_info=ldap3.OFFLINE_AD_2012_R2)
>>> mock_ldap_connection = ldap3.Connection(mock_server, client_strategy=ldap3.MOCK_SYNC, authentication=ldap3.SIMPLE)
>>> mock_ldap_connection.strategy.entries_from_json('tests/ad/ad_ldap_directory.json')
>>> mock_ldap_connection.open()
>>> mock_ldap_connection.search('DC=postmaster,DC=local', '(objectGUID=57042aa9-658f-4945-ab53-2d8bd9a1313f)', ldap3.SUBTREE)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mprahl/.virtualenvs/ldap3/lib/python2.7/site-packages/ldap3/core/connection.py", line 782, in search
    response = self.post_send_search(self.send('searchRequest', request, controls))
  File "/home/mprahl/.virtualenvs/ldap3/lib/python2.7/site-packages/ldap3/strategy/mockSync.py", line 56, in post_send_search
    responses, result = self.mock_search(request, controls)
  File "/home/mprahl/.virtualenvs/ldap3/lib/python2.7/site-packages/ldap3/strategy/mockBase.py", line 647, in mock_search
    return self._execute_search(request)
  File "/home/mprahl/.virtualenvs/ldap3/lib/python2.7/site-packages/ldap3/strategy/mockBase.py", line 677, in _execute_search
    matched = self.evaluate_filter_node(filter_root, candidates)
  File "/home/mprahl/.virtualenvs/ldap3/lib/python2.7/site-packages/ldap3/strategy/mockBase.py", line 753, in evaluate_filter_node
    self.evaluate_filter_node(element, candidates)
  File "/home/mprahl/.virtualenvs/ldap3/lib/python2.7/site-packages/ldap3/strategy/mockBase.py", line 849, in evaluate_filter_node
    if attr_name in self.connection.server.dit[candidate] and self.equal(candidate, attr_name, attr_value):
  File "/home/mprahl/.virtualenvs/ldap3/lib/python2.7/site-packages/ldap3/strategy/mockBase.py", line 863, in equal
    if self._check_equality(self._prepare_value(attribute_type, value_to_check), attribute_value):
  File "/home/mprahl/.virtualenvs/ldap3/lib/python2.7/site-packages/ldap3/strategy/mockBase.py", line 196, in _prepare_value
    validated = validator(value)
  File "/home/mprahl/.virtualenvs/ldap3/lib/python2.7/site-packages/ldap3/protocol/formatters/validators.py", line 370, in validate_uuid_le
    valid_values.append(UUID(hex=element).bytes_le)  # string representation, value in big endian, converts to little endian
  File "/usr/lib64/python2.7/uuid.py", line 136, in __init__
    raise ValueError('badly formed hexadecimal UUID string')
ValueError: badly formed hexadecimal UUID string
```

**Comment by cannatag** [comment]
This will be fixed in the next release of ldap3

**Comment by cannatag** [comment]
fixed in ldap3 v2.5.2
